test/unit/spec/MfaVerifyEmail_spec.js

```diff
@@ -0,0 +1,300 @@
+/* eslint-disable max-len */
+define([
+  'okta',
+  '@okta/okta-auth-js/jquery',
+  'util/Util',
+  'helpers/mocks/Util',
+  'helpers/dom/MfaVerifyForm',
+  'helpers/dom/Beacon',
+  'helpers/util/Expect',
+  'LoginRouter',
+  'sandbox',
+  'helpers/xhr/MFA_REQUIRED_allFactors',
+  'helpers/xhr/MFA_CHALLENGE_email',
+  'helpers/xhr/SUCCESS',
+  'helpers/xhr/MFA_LOCKED_FAILED_ATEMPTS',
+  'helpers/xhr/SMS_RESEND_error',
+], function (
+  Okta,
+  OktaAuth,
+  LoginUtil,
+  Util,
+  MfaVerifyForm,
+  Beacon,
+  Expect,
+  Router,
+  $sandbox,
+  resAllFactors,
+  resChallengeEmail,
+  resSuccess,
+  resMfaLocked,
+  resResendError,
+) {
+  var { _, $ } = Okta;
+  var itp = Expect.itp;
+
+  function createRouter (baseUrl, authClient, successSpy, settings) {
+    var router = new Router(_.extend({
+      el: $sandbox,
+      baseUrl: baseUrl,
+      authClient: authClient,
+      globalSuccessFn: successSpy
+    }, settings));
+    Util.registerRouter(router);
+    Util.mockRouterNavigate(router);
+    return router;
+  }
+
+  function setupEmail () {
+    var setNextResponse = Util.mockAjax();
+    var baseUrl = 'https://foo.com';
+    var authClient = new OktaAuth({
+      url: baseUrl,
+      transformErrorXHR: LoginUtil.transformErrorXHR
+    });
+    var successSpy = jasmine.createSpy('success');
+    var afterErrorHandler = jasmine.createSpy('afterErrorHandler');
+    var router = createRouter(baseUrl, authClient, successSpy);
+    router.on('afterError', afterErrorHandler);
+
+    setNextResponse(resAllFactors);
+    return Util.mockIntrospectResponse(router).then(function () {
+      router.refreshAuthState('dummy-token');
+      return Expect.waitForMfaVerify()
+        .then(function () {
+          var factors = router.appState.get('factors'),
+              selectedFactor = factors.findWhere({ factorType: 'email' });
+          router.verify(
+            selectedFactor.get('provider'),
+            selectedFactor.get('factorType'));
+          return Expect.wait(() => {
+            return $sandbox.find('.mfa-verify-email').length;
```

Comment:
I think we can add this to [helpers/util/Expect.js](https://github.com/okta/okta-signin-widget/blob/master/test/unit/helpers/util/Expect.js#L302), so this wait will be just something like `return Expect.waitForVerifyEmail(test);`